pkg/sriov/sriov.go

```diff
@@ -309,7 +339,48 @@ func (s *sriovManager) ApplyVFConfig(conf *sriovtypes.NetConf) error {
 		}
 	}
 
-	// 3. Set link rate
+	// 3. Set link rate.
+	if conf.MaxTxRate != nil {
+		if err = s.nLink.LinkSetVfTxRate(pfLink, conf.VFID, *conf.MaxTxRate); err != nil {
```

Comment:
FYI - this option is deprecated, from the ip tools manual:
```
rate TXRATE -- change the allowed transmit bandwidth, in Mbps, for the specified VF.  Setting this parameter to
0 disables rate limiting.  vf parameter must be specified.  Please use new API max_tx_rate option instead.
```

Reply:
Thanks, I saw this also. When looking at the netlink code for LinkSetVfTxRate(), the comment says that this is for the Max TX Bandwidth Alocation:

https://github.com/vishvananda/netlink/blob/2e4a68ee6c165085c6d708748b78cb8ac3c15ac9/nl/link_linux.go#L213

I also took a look at this during my testing. Here is the output from "ip link" when I had set the max_tx_rate to 50 Mbps:


5: em2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether ec:f4:bb:c0:bc:82 brd ff:ff:ff:ff:ff:ff
    vf 0 MAC 02:02:02:02:04:04, vlan 20, tx rate 50 (Mbps), max_tx_rate 50Mbps, spoof checking off, link-state auto, trust on, query_rss off
    vf 1 MAC 02:02:02:02:02:02, vlan 20, tx rate 50 (Mbps), max_tx_rate 50Mbps, spoof checking off, link-state auto, trust on, query_rss off
    vf 2 MAC 16:ba:a6:98:c0:fd, spoof checking on, link-state auto, trust off, query_rss off
    vf 3 MAC fe:4a:ec:08:51:ab, spoof checking on, link-state auto, trust off, query_rss off
    vf 4 MAC 26:a9:2b:2b:db:c1, spoof checking on, link-state auto, trust off, query_rss off
    vf 5 MAC 6e:e6:0d:fa:b1:2e, spoof checking on, link-state auto, trust off, query_rss off
    vf 6 MAC 4a:45:32:9e:9e:fc, spoof checking on, link-state auto, trust off, query_rss off


So, it appears this netlink call does set the max_tx_rate as specified. What is interesting, is that the "rate" attribute is also set. So, I think this is working as expected. Agree? Comments?